core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java

```diff
@@ -0,0 +1,35 @@
+package io.metadew.iesi.server.rest.encrypter;
+
+import io.metadew.iesi.server.rest.Application;
+import io.metadew.iesi.server.rest.configuration.TestConfiguration;
+import io.metadew.iesi.server.rest.configuration.security.MethodSecurityConfiguration;
+import io.metadew.iesi.server.rest.encrypter.dto.EncryptPostDto;
+import lombok.extern.log4j.Log4j2;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@Log4j2
+@SpringBootTest(classes = {Application.class, MethodSecurityConfiguration.class, TestConfiguration.class},
+        properties = {"spring.main.allow-bean-definition-overriding=true", "iesi.security.enabled=true"})
+@ExtendWith({MockitoExtension.class, SpringExtension.class})
+@ActiveProfiles({"http", "test"})
+@DirtiesContext
+class EncryptionControllerTest {
+
+    @MockBean
+    private EncryptionService encryptionService;
+
+    @MockBean
+    private EncryptionController controller;
+
+    @Test
+    void getEncryptedPassword() {
+        controller.getEncryptedPassword(new EncryptPostDto("password"));
```

Comment:
Please test in detail that controller with the method `mvc.perform`.
Also test cases when providing empty text, etc.

Reply:
For Get method we write test case with mvc.perform for post we directly invoke. Have a look on existing test classes.

Reply:
Yes, but how can you be sure if the encrypted data is encrypted properly ? we need to check that.

Reply:
We cant exactly check here because this post method, Please have a look on existing post method test cases.

Reply:
I had a look at the openApiController test, and tests case check the body response after performing post request as you can see below:
```
    @Test
    void transformFromBody() throws Exception {
        String jsonContent = new String(jsonFile);

        given(openAPIService.transform((MultipartFile) any())).willReturn(getTransformResult());
        given(transformResultDtoResourceAssembler.toModel(any())).willReturn(getTransformResultDto());

        mvc.perform(
                post("/openapi/transform")
                        .contentType(MediaType.APPLICATION_JSON)
                        .accept(MediaType.APPLICATION_JSON)
                        .content(jsonContent)
        )
                .andExpect(jsonPath("$").exists())
                .andExpect(jsonPath("$.title").isString())
                .andExpect(jsonPath("$.version").isString())
                .andExpect(jsonPath("$.connections").isArray())
                .andExpect(jsonPath("$.components").isArray())
                .andExpect(jsonPath("$.connections[0]").isMap())
                .andExpect(jsonPath("$.components[0]").isMap());
    }
```